pkg/scheduler/framework/plugins/volumezone/volume_zone_test.go

```diff
@@ -497,6 +508,117 @@ func TestWithBinding(t *testing.T) {
+				t.Errorf("PreFilter: status does not match (-want,+got):\n%s", diff)
+			}
+			filterStatus := p.Filter(ctx, state, test.Pod, node)
+			if diff := cmp.Diff(filterStatus, test.wantFilterStatus); diff != "" {
+				t.Errorf("Filter: status does not match (-want,+got):\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestWithCSI(t *testing.T) {
+	pvLister := fakeframework.PersistentVolumeLister{
+		{
+			ObjectMeta: metav1.ObjectMeta{Name: "Vol_1", Labels: map[string]string{v1.LabelFailureDomainBetaZone: "us-west1-a"}},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{Name: "Vol_2", Labels: map[string]string{v1.LabelFailureDomainBetaZone: "us-west1"}},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{Name: "Vol_CSI_1", Labels: map[string]string{v1.LabelFailureDomainBetaZone: "us-west1-a"}},
+			Spec: v1.PersistentVolumeSpec{
+				PersistentVolumeSource: v1.PersistentVolumeSource{
+					GCEPersistentDisk: &v1.GCEPersistentDiskVolumeSource{
+						PDName:    "test-disk",
+						FSType:    "ext4",
+						Partition: 0,
+						ReadOnly:  false,
+					},
+				},
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{Name: "Vol_CSI_2", Labels: map[string]string{v1.LabelFailureDomainBetaZone: "us-west1"}},
+			Spec: v1.PersistentVolumeSpec{
+				PersistentVolumeSource: v1.PersistentVolumeSource{
+					GCEPersistentDisk: &v1.GCEPersistentDiskVolumeSource{
+						PDName:    "test-disk",
+						FSType:    "ext4",
+						Partition: 0,
+						ReadOnly:  false,
+					},
+				},
+			},
+		},
+	}
+
+	pvcLister := fakeframework.PersistentVolumeClaimLister{
+		{
+			ObjectMeta: metav1.ObjectMeta{Name: "PVC_1", Namespace: "default"},
+			Spec:       v1.PersistentVolumeClaimSpec{VolumeName: "Vol_1"},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{Name: "PVC_2", Namespace: "default"},
+			Spec:       v1.PersistentVolumeClaimSpec{VolumeName: "Vol_2"},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{Name: "PVC_CSI_1", Namespace: "default"},
+			Spec:       v1.PersistentVolumeClaimSpec{VolumeName: "Vol_CSI_1"},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{Name: "PVC_CSI_2", Namespace: "default"},
+			Spec:       v1.PersistentVolumeClaimSpec{VolumeName: "Vol_CSI_2"},
+		},
+	}
+
+	tests := []struct {
+		name                string
+		Pod                 *v1.Pod
+		Node                *v1.Node
+		wantPreFilterStatus *framework.Status
+		wantFilterStatus    *framework.Status
+	}{
+		{
+			name: "migratable volume matched",
+			Pod:  createPodWithVolumes("pod_1", "PVC_CSI_1", "PVC_1"),
+			Node: &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   "host1",
+					Labels: map[string]string{v1.LabelFailureDomainBetaZone: "us-west1-a"},
+				},
+			},
+		},
+		{
+			name: "migratable volume doesn't match",
+			Pod:  createPodWithVolume("pod_1", "PVC_CSI_2", "PVC_1"),
+			Node: &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   "host1",
+					Labels: map[string]string{v1.LabelFailureDomainBetaZone: "us-west1-a"},
+				},
```

Comment:
Do we also want to cover the test case for if there is a conflict error?

Reply:
When a pv is `Migratable`, `volume_zone` plugin just skip it regardless of whether it conflicts with node's labels and will let `volumebinding` [judge].(https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/plugins/volumebinding/binder.go#L885)
So I think conflict cases are already tested in volumebinding_test, and here we just need to test if we skip the pv correctly.
We can also add a test when `volumebinding` and `volume_zone` are both enabled, but we need to add it in `schedule_one_test`. Do you think it's necessary?